modules/local/kegg_orthologs_summary.nf

```diff
@@ -9,48 +9,94 @@ process KEGG_ORTHOLOGS_SUMMARY {
-    #                adds header names, reorders the fields, and saves the result to ${prefix}_ko_summary.tsv
-    #    - stream 2: Extracts the first and second fields (KO ID and contig ID)
-    #                and saves the result to ${prefix}_ko_per_contig.tsv - this file will be be downstream in the pipeline
-    # Both TSV files are compressed with bgzip, and an index is created. The compressed files are compatible with gz, the index is
-    # used on the website.
+    # This mini-pipeline executes a series of steps to process the hmmscan output:
+    # 1. Decompresses the hmmscan_concatenated_tblout file using bgzip, which is beneficial for streaming on the website.
+    # 2. Extracts the contig ID from the protein ID format (e.g., transforms ERZxxx_1_2 to ERZxxx_1), including those from FGS.
+    # 3. Executes the hmmscan_tblout_to_tsv.py script to convert the hmmscan output to TSV format and pipes the result for further processing.
+    # 4. Uses the `tee` command to split the output into two distinct streams:
+    #    - Stream 1:
+    #      - Extracts the first (KO ID) and third (KO description) fields.
+    #      - Computes the frequency of each unique KO ID.
+    #      - Adds appropriate header names and reorders the fields.
+    #      - Saves the processed data to a file named ${prefix}_ko_summary.tsv.
+    #    - Stream 2:
+    #      - Extracts the first (KO ID) and second (contig ID) fields.
+    #      - Saves this data to a file named ${prefix}_ko_per_contig.tsv, which will be used in subsequent steps of the pipeline.
+    # Both output TSV files are compressed using bgzip, and an index is created for each.
+    # The compressed files are compatible with gz format, and the index facilitates access on the website.
 
     # If there is a retry csvtk will fail with the files already exist
-    rm -f ${prefix}_ko_per_contig.tsv.gz || true
     rm -f ${prefix}_ko_summary.tsv.gz || true
+    rm -f ${prefix}_ko_summary.tsv.gz.gzi || true
+
+    rm -f ${prefix}_ko_per_contig.tsv.gz || true
+    rm -f ${prefix}_ko_per_contig.tsv.gz || true
+
+    # Configure csvtk to use tabs
+    export CSVTK_T=true
 
-    hmmscan_tblout_to_tsv.py ${hmmscan_concatenated_tblout} | \\
+    hmmscan_tblout_to_tsv.py ${hmmscan_concatenated_tblout} | csvtk replace --no-header-row --fields 2 --pattern '^([A-Za-z]*[1-9]*_\\d+).*\$' --replacement '\$1' | \\
     tee \\
-        >(csvtk cut --num-cpus ${task.cpus} --tabs --no-header-row --fields 1,3 | \\
-            csvtk freq --num-cpus ${task.cpus} --tabs --no-header-row --fields 1,2 --reverse --sort-by-freq | \\
-            csvtk add-header --num-cpus ${task.cpus} --tabs --no-header-row --names ko,description,count | \\
-            csvtk cut --num-cpus ${task.cpus} --tabs --fields count,ko,description | \\
+        >(csvtk cut --num-cpus ${task.cpus} --no-header-row --fields 1 | \\
+            csvtk add-header --num-cpus ${task.cpus} --no-header-row --names ko | \\
```

Comment:
I think those lines are out of understanding... maybe add comments what do you do here :D what are stream 1 and what is stream 2 :fear:

Reply:
I've rewritten the "docstring", https://github.com/EBI-Metagenomics/assembly-analysis-pipeline/pull/13/commits/06a121bd05bda7eb5dde669cd44da724e2fb9afb. Does it help?

Reply:
it should explain it line by line, for both data streams coming from tee